numpyro/infer/hmc_gibbs.py

```diff
@@ -397,35 +343,42 @@ def subsample_gibbs_fn(model, model_args=(), model_kwargs={}, *, num_blocks=1):
         >>> import jax.numpy as jnp
         >>> import numpyro
         >>> import numpyro.distributions as dist
-        >>> from numpyro.infer import MCMC, NUTS, HMCGibbs, subsample_gibbs_fn
+        >>> from numpyro.infer import DiscreteHMCGibbs, MCMC, NUTS
         ...
-        >>> def model(data):
-        ...     x = numpyro.sample("x", dist.Normal(0, 1))
-        ...     with numpyro.plate("N", data.shape[0], subsample_size=100):
-        ...         batch = numpyro.subsample(data, event_dim=0)
-        ...         numpyro.sample("obs", dist.Normal(x, 1), obs=batch)
+        >>> def model(probs, locs):
+        ...     c = numpyro.sample("c", dist.Categorical(probs))
+        ...     numpyro.sample("x", dist.Normal(locs[c], 0.5))
         ...
-        >>> data = random.normal(random.PRNGKey(0), (10000,)) + 1
-        >>> gibbs_fn = subsample_gibbs_fn(model, (data,), num_blocks={'N': 10})
-        >>> kernel = HMCGibbs(NUTS(model), gibbs_fn, gibbs_sites=["N"])
-        >>> mcmc = MCMC(kernel, 1000, 1000)
-        >>> mcmc.run(random.PRNGKey(0), data)
-        >>> samples = mcmc.get_samples()["x"]
-        >>> assert abs(jnp.mean(samples).copy() - 1.) < 0.1
+        >>> probs = jnp.array([0.15, 0.3, 0.3, 0.25])
+        >>> locs = jnp.array([-2, 0, 2, 4])
+        >>> kernel = DiscreteHMCGibbs(NUTS(model), modified=True)
+        >>> mcmc = MCMC(kernel, 1000, 100000, progress_bar=False)
```

Comment:
why do this many samples if there's no `assert`?

Reply:
Oops, probably this losts during a commit. I'll reverse the assertations.